### ISSUE TITLE
Halide::select should be variadic.

### ISSUE THREAD

**Issue by kgnk** [halide/Halide#1907]
Because I needed 11 conditionals, implemented like so:

```
inline Expr select(Expr condition, Expr true_value, Expr false_value) {
    // Same as current implementation.
}

template< typename... Args >
inline Expr select(Expr c0, Expr v0, Expr c1, Expr v1, Args&&... args) {
    return select(c0, v0, select(c1, v1, std::forward<Args>(args)...));
}
```

**Comment by steven-johnson** [comment]
https://github.com/halide/Halide/pull/1919

**Comment by kgnk** [comment]
Fixed by #1919
